python/lsst/ts/externalscripts/comcam/make_comcam_bias.py

```diff
@@ -0,0 +1,157 @@
+        ----------
+        config: `types.SimpleNamespace`
+            Configuration data. See `get_schema` for information about data
+            structure.
+        """
+        # Log information about the configuration
+
+        self.log.debug(
+            f"nBias: {config.nBias}, detectors: {config.detectors}, "
+            f"instrument: {config.instrument}. "
+        )
+
+        self.config = config
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script.
+        """
+        # Temporary number
+        metadata.duration = 10
+
+    async def run(self):
+        # Should ComCam and OCPS not be enabled here in the script?
+        # original notebook says: "When writing a script, the components
+        # should (probably) be enabled by a user."
+        await self.comcam.enable()
+        await salobj.set_summary_state(self.ocps, salobj.State.ENABLED,
+                                       settingsToApply="LSSTComCam.yaml")
+
+        # Take config.nBiases biases, and return a list of IDs
+        tempBiasList = await self.comcam.take_bias(self.config.nBias)
+        exposures = tuple(tempBiasList)
+        # Bias IDs
+        await self.checkpoint(f"Biases taken: {tempBiasList}")
+
+        # did the images get archived and are they available to the butler?
+        val = await self.comcam.rem.ccarchiver.evt_imageInOODS.aget(timeout=10)
+        await self.checkpoint(f"Biases in ccarchiver: {val}")
+
+        # Now run the bias pipetask via the OCPS
+        ack = await self.ocps.cmd_execute.set_start(
+            wait_done=False, pipeline="${CP_PIPE_DIR}/pipelines/cpBias.yaml", version="",
+            config=f"-j 8 -i {self.config.inputCollections} --register-dataset-types -c isr:doDefect=False",
+            data_query=f"instrument='{self.config.instrument}' AND"
+                       f" detector IN {self.config.detectors} AND exposure IN {exposures}"
+        )
+        if ack.ack != salobj.SalRetCode.CMD_ACK:
+            ack.print_vars()
+
+        # Wait for the in-progress acknowledgement with the job identifier.
+        ack = await self.ocps.cmd_execute.next_ackcmd(ack, wait_done=False)
+        self.log.debug('Received acknowledgement of ocps command')
+
+        ack.print_vars()
+        job_id = json.loads(ack.result)["job_id"]
+
+        # Wait for the command completion acknowledgement.
+        ack = await self.ocps.cmd_execute.next_ackcmd(ack)
+        self.log.debug('Received command completion acknowledgement from ocps')
+        if ack.ack != salobj.SalRetCode.CMD_COMPLETE:
+            ack.print_vars()
+        # Wait for the job result message that matches the job id we're
+        # interested in ignoring any others (from other remotes).
+        # This obviously needs to follow the first acknowledgement
+        # (that returns the, job id) but might as well wait for the second.
+        while True:
+            msg = await self.ocps.evt_job_result.next(flush=False)
```

Comment:
Doesn't seems like you are doing anything with this message afterwards. Do you really need to wait for this event from the OCPS? If so, you may want to add a timeout here or this can block indefinitely.

Reply:
I added a timeout=600.